hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -1531,50 +1533,79 @@ public void pngPrintMutableGraph(String filePath, GraphType graphType)
   }
 
-  private List<CompactionFileInfo> toFileInfoList(
-      List<String> sstFiles,
-      ManagedOptions options,
-      ManagedReadOptions readOptions
+  /**
+   * Returns true is compactionFileInfo has column family information and
+   * doesn't belong to any of the COLUMN_FAMILIES_TO_TRACK_IN_DAG.
+   * Otherwise, false and should be added compaction DAG.
+   * CompactionFileInfo doesn't have column family information in two cases,
+   * i). Backward compatibility. Before HDDS-8940, column family is not
+   *     persisted.
+   * ii). In case, it fails to read SST in {@link this.toFileInfo()}.
+   */
+  @VisibleForTesting
+  boolean shouldSkipFile(CompactionFileInfo compactionFileInfo) {
+    return Objects.nonNull(compactionFileInfo.getColumnFamily()) &&
+        !COLUMN_FAMILIES_TO_TRACK_IN_DAG.contains(
+            compactionFileInfo.getColumnFamily());
+  }
+
+  private List<CompactionFileInfo> toFileInfoList(List<String> sstFiles,
+                                                  ManagedOptions options,
+                                                  ManagedReadOptions readOptions
   ) {
     if (CollectionUtils.isEmpty(sstFiles)) {
       return Collections.emptyList();
     }
 
-    final int fileNameOffset = sstFiles.get(0).lastIndexOf("/") + 1;
     List<CompactionFileInfo> response = new ArrayList<>();
 
     for (String sstFile : sstFiles) {
-      String fileName = sstFile.substring(fileNameOffset,
-          sstFile.length() - SST_FILE_EXTENSION_LENGTH);
-      CompactionFileInfo.Builder fileInfoBuilder =
-          new CompactionFileInfo.Builder(fileName);
-      SstFileReader fileReader = new SstFileReader(options);
-      try {
-        fileReader.open(sstFile);
-        String columnFamily = StringUtils.bytes2String(
-            fileReader.getTableProperties().getColumnFamilyName());
-        SstFileReaderIterator iterator = fileReader.newIterator(readOptions);
-        iterator.seekToFirst();
-        String startKey = StringUtils.bytes2String(iterator.key());
-        iterator.seekToLast();
-        String endKey = StringUtils.bytes2String(iterator.key());
-        fileInfoBuilder.setStartRange(startKey)
-            .setEndRange(endKey)
-            .setColumnFamily(columnFamily);
-      } catch (RocksDBException rocksDBException) {
-        // Ideally it should not happen. If it does just log the exception.
-        // And let the compaction complete without the exception.
-        // Throwing exception in compaction listener could fail the RocksDB.
-        // In case of exception, compaction node will be missing start key,
-        // end key and column family. And it will continue the traversal as
-        // it was before HDDS-8940.
-        LOG.warn("Failed to read SST file: {}.", sstFile, rocksDBException);
+      CompactionFileInfo fileInfo = toFileInfo(sstFile, options, readOptions);
+      if (shouldSkipFile(fileInfo)) {
+        LOG.debug("Skipping sst file: '{}' belongs to column family '{}'.",
```

Comment:
You would be skipping the entire compaction, you need not log for individual log file instead say skipping input files & output files are this.